src/Discovery/DiscoveryActionBar.tsx

```diff
@@ -27,8 +29,97 @@ interface Props {
   filtersVisible: boolean;
   setFiltersVisible: (boolean) => void;
   user: User,
+  downloadInProgress: boolean,
+  setDownloadInProgress: (boolean) => void,
+  downloadStatusMessage: {
+    title: string,
+    message: string,
+    active: boolean,
+    url: string
+  },
+  setDownloadStatusMessage: (Object) => void;
 }
 
+const handleDownloadZipClick = async (
+  selectedResources: any[],
+  setDownloadInProgress: (boolean) => void,
+  setDownloadStatusMessage: (object) => void,
+) => {
+  const studyIDs = selectedResources.map((study) => study.project_number);
+  setDownloadInProgress(true);
+  setDownloadStatusMessage({
+    title: 'Your download is being prepared',
+    message: 'Please remain on this page while your download is being prepared.\n\n'
+               + 'When your download is ready, it will begin automatically. You can close this window.',
+    active: true,
+    url: '',
+  });
+
+  const handleJobError = (err) => {
+    console.error(err);
+    setDownloadInProgress(false);
+    setDownloadStatusMessage({
+      title: 'Download failed',
+      message: 'There was a problem preparing your download.'
+                + 'Please consider using the gen3 client to download these files via a manifest.',
+      active: true,
+      url: '',
+    });
+  };
+
+  fetchWithCreds({
+    path: `${jobAPIPath}dispatch`,
+    method: 'POST',
+    body: JSON.stringify({
+      action: 'batch-export',
+      input: {
+        study_ids: studyIDs,
+      },
+    }),
+  }).then(
+    (dispatchResponse) => {
+      const { uid } = dispatchResponse.data;
+      const pollForJobStatusUpdate = () => {
+        fetchWithCreds({ path: `${jobAPIPath}status?UID=${uid}` }).then(
+          (statusResponse) => {
+            const { status } = statusResponse.data;
+            if (status === 'Failed') {
+              fetchWithCreds({ path: `${jobAPIPath}output?UID=${uid}` }).then(
+                (outputResponse) => {
+                  setDownloadStatusMessage({
+                    title: 'Download failed',
+                    message: outputResponse.data.output,
```

Comment:
is `data` always available in `outputResponse`? if not, this line might causing a crash

Reply:
yes, this should be guaranteed by the `fetchWithCreds` function:  https://github.com/uc-cdis/data-portal/blob/275cff44c31565020ab699a819d916342a34bab5/src/actions.js#L80